docs/use-cases/microfrontends.md

```diff
@@ -0,0 +1,308 @@
+
+| Microfrontend | Access          | `bucketBy` |
+|---------------|-----------------|------------|
+| `signup`      | Anonymous users | `deviceId` |
+| `signin`      | Anonymous users | `deviceId` |
+| `account`     | Authenticated   | `userId`   |
+
+But what shall we do for microfrontends that are accessible to both anonymous and authenticated users? Like `products` and `checkout` in our example.
+
+### Design decision
+
+It's a design decision that we need to take here when defining our features.
+
+We can either choose to:
+
+- always use `deviceId` attribute for `bucketBy`, irrespective of whether the user is anonymous or authenticated, or
+- we can choose to use `deviceId` for anonymous users and `userId` for authenticated users.
+
+The drawback of using `deviceId` at all times for both anonymous and authenticated users is that it will result in inconsistent variations for logged in users across multiple sessions or devices.
+
+If we wish to get the maximum benefit of Featurevisor's consistent bucketing making sure the same logged in user sees the same variation across all devices and sessions, we can use:
+
+- `userId` for authenticated users, and
+- `deviceId` for anonymous users
+
+### Bucket by first available attribute
+
+We can express that in our feature as follows:
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - products
+  - checkout
+
+bucketBy:
+  # if `userId` is available then it will be used,
+  # otherwise it will fall back to `deviceId`
+  or:
+    - userId
+    - deviceId
+
+# ...
+```
+
+This will make sure when `userId` attribute is passed for evaluation to the SDK, it will be used for bucketing. Otherwise, it will fall back to `deviceId`.
+
+## Review and approval workflow
+
+Given Featurevisor is a centralized feature management solution, it can help you manage your features in a microfrontends architecture in a single place conveniently.
+
+In our case, it is a single Git repository that contains all the features and their configurations, which will go through a review and approval workflow by all relevant teams before they can be deployed in the form of generated [datafiles](/docs/building-datafiles).
+
+## Consuming datafiles in your microfrontend
+
+Once you have [built](/docs/building-datafiles) and [deployed](/docs/deployment) your datafiles, you can consume them using Featurevisor SDKs in your microfrontends:
+
+```js
+// in `products` microfrontend
+import { createInstance } from "@featurevisor/sdk";
+
+const sdk = createInstance({
+  datafileUrl: "https://cdn.yoursite.com/production/datafile-tag-products.json",
+});
+```
+
+Evaluate your features:
+
+```js
+const attributes = {
+  deviceId: "...",
+
+  // if available
+  userId: "...",
+};
+
+const showMarketingBanner = sdk.getVariation(
+  "showMarketingBanner",
+  attributes
```

Comment:
100% sure this is not the right place where to ask but, I haven't found in the Docs a way to send attributes separately from retrieving a variation. Something like: the user logs in, and we set the `userId` attribute straight away. Next time someone calls `getVariation` those values are already there, like context.

Reply:
one approach would be to use the `interceptAttributes` option: https://featurevisor.com/docs/sdks/#intercepting-attributes

example:

```js
import { createInstance } from "@featurevisor/sdk";

const defaultAttributes = {
  country: "nl",
};

const sdk = createInstance({
  // ...
  interceptAttributes: function (attributes) {
    // return updated attributes
    return {
      ...defaultAttributes,
      ...attributes,
    };
  }
});
```

based on your comment, it could be like this:

```js
// after user logs in
defaultAttributes.userId = loggedInUserId;
```